lib/types/library/LibraryFormatter.js

```diff
@@ -289,7 +286,11 @@ class LibraryFormatter extends AbstractUi5Formatter {
 			throw new Error(`Given file system path ${posixFsPath} is not based on source base ` +
 				`path ${posixBasePath}.`);
 		}
-		const namespacePath = posixFsPath.replace(basePathPrefixRegExp, "");
+		let namespacePath = posixFsPath.replace(posixBasePath, "");
+		if (namespacePath.startsWith("/")) {
+			// Remove any leading slash
+			namespacePath = namespacePath.substr(1);
+		}
```

Comment:
What about the else case, isn't it an error? Depends on the question whether `posixBasePath` ends with a slash (then it's not an error) or not.

Reply:
Since we don't enforce the presence (or absence) of a trailing slash in the project path mapping, we need to deal with both cases. I updated my initial commit to deal with optional trailing slashes everywhere and make the comparison somewhat easier to understand